File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/ValueSet.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class ValueSet : DomainResource, IResource
{
    public string? Description { get; set; }
    public ValueSet_Compose? Compose { get; set; }
    public string? Date { get; set; }
    public string? Publisher { get; set; }
    public Base.CodeableConcept[]? Jurisdiction { get; set; }
    public string? Purpose { get; set; }
    public string? Name { get; set; }
    public Base.UsageContext[]? UseContext { get; set; }
    public string? Copyright { get; set; }
    public bool? Experimental { get; set; }
    public ValueSet_Expansion? Expansion { get; set; }
    public string? Title { get; set; }
    public required string Status { get; set; }
    public string? Url { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public bool? Immutable { get; set; }
    public string? Version { get; set; }
    public Base.ContactDetail[]? Contact { get; set; }

    public class ValueSet_Compose_Exclude_Filter : BackboneElement
    {
        public required string Op { get; set; }
        public required string Value { get; set; }
        public required string Property { get; set; }
    }

    public class ValueSet_Compose_Exclude_Concept_Designation : BackboneElement
    {
        public Base.Coding? Use { get; set; }
        public required string Value { get; set; }
        public string? Language { get; set; }
    }

    public class ValueSet_Compose_Exclude_Concept : BackboneElement
    {
        public required string Code { get; set; }
        public string? Display { get; set; }
        public ValueSet_Compose_Exclude_Concept_Designation[]? Designation { get; set; }
    }

    public class ValueSet_Compose_Exclude : BackboneElement
    {
        public ValueSet_Compose_Exclude_Filter[]? Filter { get; set; }
        public string? System { get; set; }
        public ValueSet_Compose_Exclude_Concept[]? Concept { get; set; }
        public string? Version { get; set; }
        public string[]? ValueSet { get; set; }
    }

    public class ValueSet_Compose_Include_Filter : BackboneElement
    {
        public required string Op { get; set; }
        public required string Value { get; set; }
        public required string Property { get; set; }
    }

    public class ValueSet_Compose_Include_Concept_Designation : BackboneElement
    {
        public Base.Coding? Use { get; set; }
        public required string Value { get; set; }
        public string? Language { get; set; }
    }

    public class ValueSet_Compose_Include_Concept : BackboneElement
    {
        public required string Code { get; set; }
        public string? Display { get; set; }
        public ValueSet_Compose_Include_Concept_Designation[]? Designation { get; set; }
    }

    public class ValueSet_Compose_Include : BackboneElement
    {
        public ValueSet_Compose_Include_Filter[]? Filter { get; set; }
        public string? System { get; set; }
        public ValueSet_Compose_Include_Concept[]? Concept { get; set; }
        public string? Version { get; set; }
        public string[]? ValueSet { get; set; }
    }

    public class ValueSet_Compose : BackboneElement
    {
        public ValueSet_Compose_Exclude[]? Exclude { get; set; }
        public required ValueSet_Compose_Include[] Include { get; set; }
        public bool? Inactive { get; set; }
        public string? LockedDate { get; set; }
    }

    public class ValueSet_Expansion_Contains_Contains : BackboneElement
    {
        public string? Code { get; set; }
        public string? System { get; set; }
        public string? Display { get; set; }
        public string? Version { get; set; }
        public bool? Abstract { get; set; }
        public string[]? Contains { get; set; }
        public bool? Inactive { get; set; }
        public string[]? Designation { get; set; }
    }

    public class ValueSet_Expansion_Contains_Designation : BackboneElement
    {
        public Base.Coding? Use { get; set; }
        public required string Value { get; set; }
        public string? Language { get; set; }
    }

    public class ValueSet_Expansion_Contains : BackboneElement
    {
        public string? Code { get; set; }
        public string? System { get; set; }
        public string? Display { get; set; }
        public string? Version { get; set; }
        public bool? Abstract { get; set; }
        public ValueSet_Expansion_Contains_Contains[]? Contains { get; set; }
        public bool? Inactive { get; set; }
        public ValueSet_Expansion_Contains_Designation[]? Designation { get; set; }
    }

    public class ValueSet_Expansion_Parameter : BackboneElement
    {
        public string? ValueCode { get; set; }
        public string? ValueUri { get; set; }
        public string? ValueDecimal { get; set; }
        public required string Name { get; set; }
        public object? Value    
        {
            get
            {
                if (ValueCode is not null)
                {
                    return ValueCode;
                }
        
                if (ValueUri is not null)
                {
                    return ValueUri;
                }
        
                if (ValueDecimal is not null)
                {
                    return ValueDecimal;
                }
        
                if (ValueString is not null)
                {
                    return ValueString;
                }
        
                if (ValueBoolean is not null)
                {
                    return ValueBoolean;
                }
        
                if (ValueDateTime is not null)
                {
                    return ValueDateTime;
                }
        
                if (ValueInteger is not null)
                {
                    return ValueInteger;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(string))
                {
                    ValueCode = (string)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueUri = (string)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueDecimal = (string)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueString = (string)value;return;
                }
        
                if (value?.GetType() == typeof(bool))
                {
                    ValueBoolean = (bool)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ValueDateTime = (string)value;return;
                }
        
                if (value?.GetType() == typeof(int))
                {
                    ValueInteger = (int)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public string? ValueString { get; set; }
        public bool? ValueBoolean { get; set; }
        public string? ValueDateTime { get; set; }
        public int? ValueInteger { get; set; }
    }

    public class ValueSet_Expansion : BackboneElement
    {
        public int? Total { get; set; }
        public int? Offset { get; set; }
        public ValueSet_Expansion_Contains[]? Contains { get; set; }
        public ValueSet_Expansion_Parameter[]? Parameter { get; set; }
        public required string Timestamp { get; set; }
        public string? Identifier { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/ValueSet.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import UsageContext
from base import ContactDetail
from base import CodeableConcept
from base import Coding
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class ValueSetComposeIncludeFilter(BackboneElement):
    op: str
    value: str
    property: str

@dataclass(kw_only=True)
class ValueSetComposeIncludeConceptDesignation(BackboneElement):
    use: Optional[Coding] = None
    value: str
    language: Optional[str] = None

@dataclass(kw_only=True)
class ValueSetComposeIncludeConcept(BackboneElement):
    code: str
    display: Optional[str] = None
    designation: Optional[List[ValueSetComposeIncludeConceptDesignation]] = field(default_factory=list)

@dataclass(kw_only=True)
class ValueSetComposeInclude(BackboneElement):
    filter: Optional[List[ValueSetComposeIncludeFilter]] = field(default_factory=list)
    system: Optional[str] = None
    concept: Optional[List[ValueSetComposeIncludeConcept]] = field(default_factory=list)
    version: Optional[str] = None
    value_set: Optional[List[str]] = field(default_factory=list)

@dataclass(kw_only=True)
class ValueSetCompose(BackboneElement):
    exclude: Optional[List[Reference]] = field(default_factory=list)
    include: list[ValueSetComposeInclude] = field(default_factory=list)
    inactive: Optional[bool] = None
    locked_date: Optional[str] = None

@dataclass(kw_only=True)
class ValueSetExpansionContains(BackboneElement):
    code: Optional[str] = None
    system: Optional[str] = None
    display: Optional[str] = None
    version: Optional[str] = None
    abstract: Optional[bool] = None
    contains: Optional[List[Reference]] = field(default_factory=list)
    inactive: Optional[bool] = None
    designation: Optional[List[Reference]] = field(default_factory=list)

@dataclass(kw_only=True)
class ValueSetExpansionParameter(BackboneElement):
    value_code: Optional[str] = None
    value_uri: Optional[str] = None
    value_decimal: Optional[float] = None
    name: str
    value_string: Optional[str] = None
    value_boolean: Optional[bool] = None
    value_date_time: Optional[str] = None
    value_integer: Optional[int] = None

@dataclass(kw_only=True)
class ValueSetExpansion(BackboneElement):
    total: Optional[int] = None
    offset: Optional[int] = None
    contains: Optional[List[ValueSetExpansionContains]] = field(default_factory=list)
    parameter: Optional[List[ValueSetExpansionParameter]] = field(default_factory=list)
    timestamp: str
    identifier: Optional[str] = None

@dataclass(kw_only=True)
class ValueSet(DomainResource):
    description: Optional[str] = None
    compose: Optional[ValueSetCompose] = None
    date: Optional[str] = None
    publisher: Optional[str] = None
    jurisdiction: Optional[List[CodeableConcept]] = field(default_factory=list)
    purpose: Optional[str] = None
    name: Optional[str] = None
    use_context: Optional[List[UsageContext]] = field(default_factory=list)
    copyright: Optional[str] = None
    experimental: Optional[bool] = None
    expansion: Optional[ValueSetExpansion] = None
    title: Optional[str] = None
    status: str
    url: Optional[str] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    immutable: Optional[bool] = None
    version: Optional[str] = None
    contact: Optional[List[ContactDetail]] = field(default_factory=list)